### DISCUSSION TITLE
Specifying a default publish_mode value

### DISCUSSION THREAD

**Discussion by privatemaker** [decaporg/decap-cms#7095] [Q&A] [answered]
Following and testing the Gitea support discussed in #6984, #6997 it seems now that Decap works on Gitea and the [Forgejo](https://codeberg.org/forgejo/forgejo) fork when the server is configured correctly, which is very cool 🥳 

However, while testing a couple sites, I noticed the following error when using the nifty Hugo Module [hugomods/decap-cms](https://github.com/hugomods/decap-cms) made by @razonyang

![Screenshot 2024-02-07 at 18-26-15 Editor](https://github.com/decaporg/decap-cms/assets/74658385/65f5568f-390e-4573-a5dd-8791622c5303)

Upon inspecting the `config.yml` file the module generates, which is served as `decap-cms-confg.yml` it seems the module is injecting `publish_mode: editorial_workflow` even when it's not specified. However, since with `gitea` this is incompatible, I tried changing value to *Boolean* `false` which shows:

```
'publish_mode' must be string
'publish_mode' must be equal to one of the allowed values
```

Also when setting it as *String* values `"false"` or `"default"` or `"normal"` it still fails with this error:

```
'publish_mode' must be equal to one of the allowed values
```

I can think of the following ways to resolve this issue and use the Hugo Module with Gitea:

- The [Decap docs](https://decapcms.org/docs/configuration-options/#publish-mode) are lacking what value is considered "normal" publish mode or I am blinding missing it.
- Decap CMS code needs improvement allowing for specifying values other than `editorial_workflow` 
- @razonyang needs to modify his module to not automatically inject the `publish_mode: editorial_workflow` but rather with whatever default or normal value is.

**Comment by razonyang** [comment, marked as answer]
@privatemaker I think the issue is belong to that Hugo module, now it should be fixed since [github.com/hugomods/decap-cms@v0.13.1](https://github.com/hugomods/decap-cms/releases/tag/v0.13.1). To remove the default `publish_mode` from that module.

```yaml
params:
  decap_cms:
    publish_mode: ''
```

**Reply by privatemaker** [reply]
@razonyang thanks for quick patch. That solves a few headaches on my end 😉 I will mark your setting as an answer as it makes Decap + Hugo Mod + Gitea work now 🎉 

I still think this should be considered an dev bug in Decap to treat null value `""` as representative of one significant UX flow vs. another. If @martinjagodic agrees, maybe this should be turned into an issue.

**Reply by martinjagodic** [reply]
Yes, this is worthy of an issue
